compiler/rustc_parse/src/parser/pat.rs

```diff
@@ -315,7 +315,18 @@ impl<'a> Parser<'a> {
             PatKind::Box(pat)
         } else if self.check_inline_const() {
             // Parse `const pat`
-            PatKind::Lit(self.parse_const_expr(lo.to(self.token.span))?)
+            let const_expr = self.parse_const_expr(lo.to(self.token.span))?;
+
+            if let Some(re) = self.parse_range_end() {
+                if !self.check(&token::FatArrow) {
```

Comment:
I think `parse_pat_range_begin_with` is supposed to be used here.
See how range end is treated in other cases (`CONST .. <END>` and `10 .. <END>`).